homeassistant/components/droplet/sensor.py

```diff
@@ -0,0 +1,117 @@
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pydroplet.droplet import Droplet
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import EntityCategory, UnitOfVolume, UnitOfVolumeFlowRate
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    KEY_CURRENT_FLOW_RATE,
+    KEY_SERVER_CONNECTIVITY,
+    KEY_SIGNAL_QUALITY,
+    KEY_VOLUME,
+)
+from .coordinator import DropletConfigEntry, DropletDataCoordinator
+
+
+@dataclass(kw_only=True, frozen=True)
+class DropletSensorEntityDescription(SensorEntityDescription):
+    """Describes Droplet sensor entity."""
+
+    value_fn: Callable[[Droplet], float | str | None]
+
+
+SENSORS: list[DropletSensorEntityDescription] = [
+    DropletSensorEntityDescription(
+        key=KEY_CURRENT_FLOW_RATE,
+        translation_key=KEY_CURRENT_FLOW_RATE,
+        device_class=SensorDeviceClass.VOLUME_FLOW_RATE,
+        native_unit_of_measurement=UnitOfVolumeFlowRate.LITERS_PER_MINUTE,
+        suggested_unit_of_measurement=UnitOfVolumeFlowRate.GALLONS_PER_MINUTE,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda device: device.get_flow_rate(),
+    ),
+    DropletSensorEntityDescription(
+        key=KEY_VOLUME,
+        translation_key=KEY_VOLUME,
+        device_class=SensorDeviceClass.WATER,
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        suggested_unit_of_measurement=UnitOfVolume.GALLONS,
+        suggested_display_precision=2,
+        state_class=SensorStateClass.TOTAL,
+        value_fn=lambda device: device.get_volume_delta(),
+    ),
+    DropletSensorEntityDescription(
+        key=KEY_SERVER_CONNECTIVITY,
+        translation_key=KEY_SERVER_CONNECTIVITY,
+        device_class=SensorDeviceClass.ENUM,
+        options=["connected", "connecting", "disconnected"],
+        value_fn=lambda device: device.get_server_status(),
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    DropletSensorEntityDescription(
+        key=KEY_SIGNAL_QUALITY,
+        translation_key=KEY_SIGNAL_QUALITY,
+        device_class=SensorDeviceClass.ENUM,
+        options=["no_signal", "weak_signal", "strong_signal"],
+        value_fn=lambda device: device.get_signal_quality(),
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: DropletConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Droplet sensors from config entry."""
+    coordinator: DropletDataCoordinator = config_entry.runtime_data
```

Comment:
```suggestion
    coordinator = config_entry.runtime_data
```
No need to overwrite the type as the runtime data already is typed